File: MainApp/Renting/src/main/java/jvn/Renting/model/RentInfo.java
package jvn.Renting.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class RentInfo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDateTime dateTimeFrom;

    @Column(nullable = false)
    private LocalDateTime dateTimeTo;

    @Column(nullable = false)
    private Long advertisement;

    @Column
    private Boolean optedForCDW;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private RentRequest rentRequest;

    @OneToOne(mappedBy = "rentInfo", cascade = CascadeType.ALL)
    private RentReport rentReport;

    @Column
    private Integer rating;

    @OneToMany(mappedBy = "rentInfo", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Set<Comment> comments = new HashSet<>();

    @Column
    private Integer kilometresLimit;

    @Column
    private Double pricePerKm;
}

File: MainApp/ZuulAndFrontend/Frontend/src/app/model/rentInfo.ts
import { AdvertisementFromSearch } from "./advertisementFromSearch";
import { RentReport } from "./rentReport";
import { Comment } from "./comment";

export class RentInfo {
  id: number;
  dateTimeFrom: string;
  dateTimeTo: string;
  optedForCDW: boolean;
  advertisement: AdvertisementFromSearch;
  inBundle: boolean;
  index: number;
  additionalCost: number;
  paid: boolean;
  comments: Comment[];
  rentReport: RentReport;
  rating: number;
  constructor(
    dateTimeFrom: string,
    dateTimeTo: string,
    optedForCDW: boolean,
    advertisement: AdvertisementFromSearch,
    comments?: Comment[],
    id?: number,
    additionalCost?: number,
    paid?: boolean,
    rentReport?: RentReport,
    rating?: number
  ) {
    this.dateTimeFrom = dateTimeFrom;
    this.dateTimeTo = dateTimeTo;
    this.optedForCDW = optedForCDW;
    this.advertisement = advertisement;
    this.comments = comments;
    this.id = id;
    this.inBundle = false;
    this.additionalCost = additionalCost;
    this.paid = paid;
    this.rentReport = rentReport;
    this.rating = rating;
  }
}
